### ISSUE TITLE
Could not find package in a version matching...

### ISSUE THREAD

**Issue by o5** [composer/composer#10025]
Output of `composer diagnose`:

```bash
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys:
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: 2.1.5
PHP version: 7.4.21
PHP binary path: /usr/local/bin/php
OpenSSL version: OpenSSL 1.1.1k  25 Mar 2021
cURL version: 7.77.0 libz 1.2.11 ssl OpenSSL/1.1.1k
zip: extension not loaded, unzip present, 7-Zip not available
```

When I run this command: 

```bash
composer global require slimapi/coding-standard=0.6.0
```

I get the following output: 

```bash
Changed current directory to /root/.composer


  [InvalidArgumentException]
  Could not find package slimapi/coding-standard in a version matching "0.6.0" and a stability matching "dev".


require [--dev] [--dry-run] [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--fixed] [--no-suggest] [--no-progress] [--no-update] [--no-install] [--no-scripts] [--update-no-dev] [-w|--update-with-dependencies] [-W|--update-with-all-dependencies] [--with-dependencies] [--with-all-dependencies] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [--sort-packages] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--] [<packages>]...
```

And I expected this to happen:

* Version with tag [`v0.6.0`](https://packagist.org/packages/slimapi/coding-standard) should be installed
* Tried `composer clearcache && composer cc` but it did not help
* If I do not specify version, 0.5.0 will be installed

**EDIT**: Tried to publish a new version (tag 0.6.1) but still same :/

**Comment by camohub** [comment]
I have the same problem with versions with simple command composer require xxx/xxx:1.2.0 Composer is able to find older versions but not the few latest. It stop on version 1.0.29 which can be downloaded. But version 1.0.30 and above throws me an error: 

>  [InvalidArgumentException]
>   Could not find package camohub/laravel-datagrid in a version matching 1.2.0

One hour ago everything worked fine.

**Comment by o5** [comment]
Probably something with packagist.org??

**Comment by camohub** [comment]
Packagist can see the new versions. But composer can not download them.

**Comment by o5** [comment]
Tried to publish a new version (tag 0.6.1) but still same :/

**Comment by o5** [comment]
The same for `dev-master` (composer downloaded and old VCS hash)

**Comment by daneren2005** [comment]
I am having the same issue with https://packagist.org/packages/daneren2005/php-resque.  Nothing I do seems to get it to see 1.2.30 or 1.2.31.  I even tried creating the new tag for 1.2.31 just to see if that fixed it and composer still sees nothing but 1.2.29 when I use `composer show -a daneren2005/php-resque`.  I tried clearing cache, completely wiping vendor dir, updating to latest composer release, etc... and it is still borked.

**Comment by Seldaek** [comment]
It's back now, sorry we had some major failure on the metadata dumping which I didn't pick up quickly enough due to bad combination of vacation and missing monitoring on something.
